Sources/StreamChat/Utils/Operations/AsyncOperation.swift

```diff
@@ -15,7 +15,7 @@ class AsyncOperation: BaseOperation {
     private var executedRetries = 0
 
     var canRetry: Bool {
-        executedRetries < maxRetries && !isCancelled
+        executedRetries < maxRetries && !isCancelled && !isFinished
```

Comment:
Looks inoffensive but this actually created an edge case where we would be calling the completion of an API request up to 10 times at the SAME exact moment, leading to really bad data races.

Reply:
Can we add a comment explaining that then? Someone might remove this since it looks inoffensive 🤔